### PR TITLE
#12852 - Show layers based on user rights

### LINKED ISSUE #12852: Remove from 'Layers' the options related to entities for which the user does not have at least view rights over [1]
### Problem Description

On the Dashboard maps, in the 'Layers' section, there are options that are displayed even when the user does not have rights to see the respective entities. 

One scenario is that when the user does not have rights over contacts, and they access the Surveillance Dashboard -> Case Status Map -> Layers -> Show contacts - they get an error as they do not have rights over that type of entity.

### Proposed Change

When a user does not have view rights over an entity, remove it from the Dashboard Maps Layers options.

### Added Value/Benefit

Consistency in the display of data.

### Acceptance Criteria

- [x] in the 'Layers' of maps on the Dashboards (Surveillance, Contacts, Samples) remove the options that are associated with entities over which the user does not have view rights over; 
- [x] on the 'Surveillance' Dashboard, if the user does not have view rights for contacts, remove the section referring to contacts from the layers;
- [x] on the 'Contacts' Dashboard, if the user does not have view rights over events, remove the events entry from the layers;
- [x] on the 'Samples' Dashboard, if the user does not have view rights for any one of the entities in the layers section (case, contacts, events, environments) remove the entry from the layers ( ex. for consistency: in the 'Samples' directory, if the rights to view one entity is missing - contact - the samples associated with contacts are not displayed and neither is the column in the main list).

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by MateStrysewske**
Fixes #12852

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12942

**Comment by carina29** [comment]
@MateStrysewske  I've also tested the feature branch. 

If the user does not have any view rights from the Layers section, an empty dropdown will appear (in this scenario I've had only the Samples related rights and the Access to the Samples dashboard but probably can be reproduced with other combinations too when we have access only to one dashboard without view rights for other entities):
image

image

User Rights for my test user:
image
image
image
image
image

image

image


I'm not sure how should we proceed in this case : Disable the button, hide the button or maybe you have other ideas?

File: sormas-ui/src/main/java/de/symeda/sormas/ui/dashboard/map/DashboardMapComponent.java [hunk 1]
```diff
@@ -259,88 +260,97 @@ protected void addLayerOptions(VerticalLayout layersLayout) {
 			refreshMap(true);
 		});
```
Comment (carina29) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/dashboard/map/DashboardMapComponent.java:262]: I've added a comment to this PR.

**Review by carina29 — CHANGES_REQUESTED** [review]

**Review by carina29 — APPROVED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12942

**Pull request merged**